### DISCUSSION TITLE
What's the recommended way to configure RESTHeart with docker-compose?

### DISCUSSION THREAD

**Discussion by jonathanthiry** [SoftInstigate/restheart#434] [Q&A] [answered]
Hi!

I'm using docker-compose and would like to override `mongo-mounts` in restheart.yml. I'm mounting my modified restheart.yml as a volume, but I'll have to make sure to update the file and reapply my changes every time I upgrade RESTHeart.

Is this the recommended way to it, or is there a way to only override some property in a separate file?

Thanks!

**Comment by mkjsix** [comment, marked as answer]
Hi,

In this case, I suggest adopting a SNAPSHOT version of RESTHeart built from `master` branch. Our Docker images are available at https://hub.docker.com/r/softinstigate/restheart-snapshot 

Then you can use the `environment` directive in your docker-compose.yml to override the new `RHO` environment variable. This would require a [docker-compose restart](https://docs.docker.com/compose/reference/restart/) command.

The reason is that RESTHeart v6 allows overriding configuration options with an environment variable **only if the parameters are a primitive type**.

Taking into account the following configuration snipped:

```yaml
mongo-uri: mongodb://127.0.0.1
mongo-mounts:
  - what: restheart
    where: /
```

The env var RH_MONGO_URI overrides the parameter mongo-uri.
The parameters mongo-mounts, not being of a primitive type, cannot be overridden.

RESTHeart v7 introduces a new way to override variables via the environment variable RHO.

An example is

`$ RHO='/mongo-uri->"mongodb://127.0.0.1";/mongo-mounts[1]/where->"/api"' java -jar restheart.jar`


or even

`$ RHO='/mongo-uri->"mongodb://127.0.0.1";/mongo-mounts[1]->{"where: "/api", "what": "mydb"}' java -jar restheart.jar`

**Reply by jonathanthiry** [reply]
Thanks @mkjsix, I'm looking forward to v7 then. Keep up the good work!
